internal/docstore/dynamodocstore/codec.go

```diff
@@ -332,14 +341,28 @@ func toGoValue(av *dyn.AttributeValue) (interface{}, error) {
 }
 
 func (d decoder) AsSpecial(v reflect.Value) (bool, interface{}, error) {
+	unsupportedFmt := "unsupported type: %v, the docstore driver for DynamoDB does not decode set types"
 	switch v.Type() {
 	case typeOfGoTime:
 		if d.av.S == nil {
 			return false, nil, errors.New("expected string field for time.Time")
 		}
 		t, err := time.Parse(time.RFC3339Nano, *d.av.S)
 		return true, t, err
-	default:
-		return false, nil, nil
+	case typeOfBinarySet:
```

Comment:
I think what you want to do here is see if `d.av` is one of SS, NS or BS, and return an error. It shouldn't matter what `v` is.

Reply:
Done.